### DISCUSSION TITLE
Using Zen-Engine in Go

### DISCUSSION THREAD

**Discussion by ishaan812** [gorules/zen#134] [Show and tell]
I wanted to use ZenEngine in Go and self host but was disappointed to see the only way to simply integrate was to build another service and use the evaluation API. So I decided to build something that I could have in the same service. 

Go-Rust-Gorules: [ishaan812/go-rust-testing](https://github.com/ishaan812/go-rust-testing)

I compiled the rust package into C and used CGO to interface with the package. It cut down the overhead for an evaluation from ~200 ms (Evaluation API Call) to ~1 ms (In the go-c-rust call). It was a fun exercise for me and I hope this helps any other Gopher like me to do this easily if they were to stumble upon this package :)

**Comment by stefan-gorules** [comment]
Hi @ishaan812, Thanks for sharing!

You can also use https://github.com/gorules/zen-go/ which uses the methodology you implemented above and provides you with clean interface for usage in Go.

Let me know if this makes sense, not sure if you were doing something outside of these bounds. If you were would love to understand your pain point more.

**Comment by ishaan812** [comment]
Oooo, yup pretty much the same thing, I had checked and logged an issue earlier to try and integrate with Go but I got a reply that evaluation api was the only way then and this was on your roadmap, so implemented it myself a while ago. Super cool! It was a great learning experience either ways! Thanks!
